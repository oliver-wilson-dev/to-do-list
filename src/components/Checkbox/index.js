import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';

const Checkbox = ({ markComplete, id, complete }) => {
  const onChange = () => markComplete({ id });

  return (
    <input
      type="checkbox"
      className={styles.checkbox}
      checked={complete}
      onChange={onChange}
      aria-label={`mark this task as ${complete ? 'incomplete' : 'complete'}`}
    />
  );
};

Checkbox.propTypes = {
  markComplete: PropTypes.func.isRequired,
  id: PropTypes.symbol.isRequired,
  complete: PropTypes.bool.isRequired
};

export default Checkbox;
