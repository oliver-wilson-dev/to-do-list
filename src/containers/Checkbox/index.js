import { connect } from 'react-redux';
import Checkbox from '../../components/Checkbox';
import markComplete from '../../state/actions/markComplete';

const mapDispatchToProps = dispatch => ({
  markComplete:
  ({ id }) => dispatch(markComplete({ id }))
});

export default connect(undefined, mapDispatchToProps)(Checkbox);
