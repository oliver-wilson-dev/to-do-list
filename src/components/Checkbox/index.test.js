
import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from './index';

const mockMarkComplete = jest.fn();
const mockId = Symbol('test-id');

const defaultProps = { markComplete: mockMarkComplete, complete: false, id: mockId };

const render = overrideProps => shallow(<Checkbox {...defaultProps} {...overrideProps} />);

describe('<Checkbox/> component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  describe('when the checkbox is checked', () => {
    it('should render correctly', () => {
      expect(render({ complete: true })).toMatchSnapshot();
    });

    describe('and the checkbox is then unchecked', () => {
      it('should call the markComplete function with the id of the task', () => {
        render({ complete: true }).simulate('change');

        expect(mockMarkComplete).toHaveBeenCalledWith({ id: mockId });
      });
    });
  });

  describe('when the checkbox is not checked', () => {
    it('should render correctly', () => {
      expect(render()).toMatchSnapshot();
    });

    describe('and the checkbox is then checked', () => {
      it('should call the markComplete function with the id of the task', () => {
        render().simulate('change');

        expect(mockMarkComplete).toHaveBeenCalledWith({ id: mockId });
      });
    });
  });
});
