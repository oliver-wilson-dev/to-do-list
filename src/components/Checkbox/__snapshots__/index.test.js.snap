// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> component when the checkbox is checked should render correctly 1`] = `
<input
  checked={true}
  className="checkbox"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`<Checkbox/> component when the checkbox is not checked should render correctly 1`] = `
<input
  checked={false}
  className="checkbox"
  onChange={[Function]}
  type="checkbox"
/>
`;
